plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/ProjectWrapper.java

```diff
@@ -65,6 +76,15 @@ protected void postUpdate(Collection<IResource> resources, Map<IResource, Abstra
 					resources);
 			newService.initWithResources(relatedResources);
 			newWrapper = newService;
+		} else if ((ResourceKind.DEPLOYMENT_CONFIG.equals(r.getKind()) ||
+		           (ResourceKind.REPLICATION_CONTROLLER.equals(r.getKind()) && !r.isAnnotatedWith(OpenShiftAPIAnnotations.DEPLOYMENT_CONFIG_NAME))) &&
+		           ServiceResourceMapper.getServices((IReplicationController) r, resources).isEmpty()) {
+		        ReplicationControllerWrapper dcWrapper = new ReplicationControllerWrapper(this, (IReplicationController) r);
+		        Collection<IResource> relatedResource = 

```

Comment:
Code from lines [83](https://github.com/jbosstools/jbosstools-openshift/pull/1427/files#diff-d73789e8250b3ee63bf704c4edf0e588R83) and [62](https://github.com/jbosstools/jbosstools-openshift/pull/1427/files#diff-d73789e8250b3ee63bf704c4edf0e588R62) can be separated in a method. 

also please, format it, it's hard to read